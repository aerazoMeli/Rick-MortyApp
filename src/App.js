import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';
import { Grid, Col, Row } from 'react-flexbox-grid';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import CharacterContainer from './containers/CharacterContainer';
import CharacterListContainer from './containers/CharacterListContainer';

function App() {
  const [characterSelected, selectCharacter] = useState(null);

  const handleClickCharacter  = character => {
    console.log(character);
    selectCharacter(character);
  };


  const handles = {
     handleClickCharacter,
  };

  return (
    <Grid>
    <Row>
        <AppBar position="sticky">
                <Toolbar>
                    Rick & Morty App
                </Toolbar>
        </AppBar>
    </Row>
    <Row>
        <Col md={6}>
          <CharacterListContainer handles={handles}/>
        </Col>
        <Col md={6}>
          <CharacterContainer character={characterSelected}/>
        </Col>
    </Row>
</Grid>
  );
}

export default App;
