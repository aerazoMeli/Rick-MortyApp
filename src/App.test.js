import React from 'react';
// import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/Rick & Morty App/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('render component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});