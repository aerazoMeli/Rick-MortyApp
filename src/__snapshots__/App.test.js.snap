// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component should render correctly in "debug" mode 1`] = `
<Grid>
  <Row>
    <WithStyles(ForwardRef(AppBar))
      position="sticky"
    >
      <WithStyles(ForwardRef(Toolbar))>
        Rick & Morty App
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
  </Row>
  <Row>
    <Col
      md={6}
    >
      <Component
        handlers={
          Object {
            "handleClickCharacter": [Function],
          }
        }
      />
    </Col>
    <Col
      md={6}
    >
      <CharacterContainer
        character={null}
      />
    </Col>
  </Row>
</Grid>
`;
